lib/active_record/mysql/enum/mysql_adapter.rb

```diff
@@ -45,20 +45,30 @@ def type_to_sql(type, limit: nil, **_options) # :nodoc:
           end
         end
 
+        if Gem::Version.new(Rails.version.to_f.to_s) < Gem::Version.new('7.0')
+          private
 
-        private
+          def initialize_type_map(m = type_map)
+            super
+
+            m.register_type(/enum/i) do |sql_type|
+              limit = sql_type.sub(/^enum\('(.+)'\)/i, '\1').split("','").map { |v| v.to_sym }
+              ActiveRecord::Type::Enum.new(limit: limit)
+            end
+          end
+        end
+      end
 
-        def initialize_type_map(m = type_map)
-          super
+      ActiveRecordMysqlAdapter.prepend ActiveRecord::Mysql::Enum::MysqlAdapter
 
+      unless Gem::Version.new(Rails.version.to_f.to_s) < Gem::Version.new('7.0')
```

Comment:
Same request here around removing the `to_f.to_s` from the `Rails.version`

Reply:
Done